### DISCUSSION TITLE
JWT for client credentials flow not possible?

### DISCUSSION THREAD

**Discussion by 0Tom2020** [ory/hydra#2721] [Q&A] [answered]
Hello together, 

The starting point is a microservice architecture. If an end user receives a JWT, he can use it to address all microservices. And if a microservice calls up another service on behalf of a user, the user's token is transferred in this request. (Principal Propagation). Now, however, a service can also call another service without the identity of a user. So the client credentials flow is used here. However, with a client credentials flow, Hydra does not create a JWT and thus the services cannot be called.

Is this a matter of attitude or does Hydra not support this use case?

Before that I used Keycloak, it was possible there. Both a user and a pure oauth client received a JWT there.

Many thanks for your help!

**Comment by vinckr** [comment, marked as answer]
Hello @0Tom2020, 

So it is not possible in the way you are used to from Keycloak, but you can use Ory Oathkeeper to achieve the same functionality. 
Check out this document, I hope it helps to clear things up a bit more: 
https://www.ory.sh/hydra/docs/advanced/#json-web-tokens . 

Let me know if that helped!

**Reply by vinckr** [reply]
Hey @0Tom2020 ,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
